resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -145,6 +157,13 @@ private[spark] class ExecutorPodsAllocator(
       logDebug(s"Pod allocation status: $currentRunningCount running, " +
         s"${currentPendingExecutors.size} pending, " +
         s"${newlyCreatedExecutors.size} unacknowledged.")
+
+      val numFailedExecutors =
+        EXECUTOR_ID_COUNTER.get() - currentPendingExecutors.size - currentRunningCount
+      if (numFailedExecutors >= maxNumExecutorFailures) {
+        logError(s"Max number of executor failures ($maxNumExecutorFailures) reached")
+        System.exit(EXIT_MAX_EXECUTOR_FAILURES)
```

Comment:
This isn't a good way to stop the application. It may be running as part of other code that can recover from such a failure.

You should do something like stop the SparkContext instead.